app/controllers/hyrax/file_sets_controller.rb

```diff
@@ -147,11 +153,24 @@ def search_builder_class
 
     def initialize_edit_form
       @parent = @file_set.in_objects.first
+      workflow_check(parent_id: @parent.id)
       original = @file_set.original_file
       @version_list = Hyrax::VersionListPresenter.new(original ? original.versions.all : [])
       @groups = current_user.groups
     end
 
+    def workflow_check(parent_id: nil)
+      return if current_ability.can?(:edit, @solr_document)
+      if parent_id.nil?
+        curation_concern = ::FileSet.find(params[:id])
```

Comment:
i'm *slightly* worried about the cost of this lookup, and not easily understanding when it's triggered.

in what cases would i pass a `nil` argument here? could this branch be separated and called explicitly, instead of branching on the `.nil?` type check?